### PR TITLE
Fix http cookie bug

### LINKED ISSUE #24724: getCookies() in http throws 'IndexOutOfRange' for cookies with no value
**Description:**
When there is no value for cookie key, the `getCookies()` function throws `IndexOutOfRange`.
sample cookie:
```
key=;
```
or
```
key
```
Error:
```
error: {ballerina/lang.array}IndexOutOfRange message=array index out of range: index: 1, size: 1
        at ballerina.http.1_0_0.cookie.cookie:parseCookieHeader(cookie/cookie.bal:233)
           ballerina.http.1_0_0.Request:getCookies(http_request.bal:565)
```

**Steps to reproduce:**
use `getCookies()` function to read cookie header with one of the above cookies.

**Affected Versions:**
Swan Lake Preview 1

### PR DESCRIPTION
**Pull request by kalaiyarasiganeshalingam**
## Purpose
$Subject

Fixes #24724

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: stdlib/http/src/main/ballerina/src/http/cookie/cookie.bal [hunk 1]
```diff
@@ -229,9 +230,18 @@ function parseCookieHeader(string cookieStringValue) returns Cookie[] {
...
+            string[] nameValue = stringutils:split(item, EQUALS);
+            Cookie cookie;
+            if (nameValue.length() > 1) {
+                cookie = new (nameValue[0], nameValue[1]);
+            } else {
+                cookie = new (nameValue[0], "");
+            }
+            cookiesInRequest.push(cookie);
+        } else {
+            log:printError("Invalid cookie: " + item + ", which must be in the format as [{name}=].");
```
Comment (chamil321) [inline stdlib/http/src/main/ballerina/src/http/cookie/cookie.bal:243]: Shall we add a test case?

File: tests/jballerina-integration-test/src/test/resources/http/src/httpservices/43_http_cookies.bal [hunk 2]
```diff
@@ -258,4 +258,23 @@ service cookie on new http:Listener(9253) {
             var result = caller->respond(res);
         }
     }
+
+    @http:ResourceConfig {
+        methods: ["GET"],
+        path: "/cookieBackend_13"
+   }
+    resource function validateCookie(http:Caller caller, http:Request req) {
+        http:Cookie[] reqstCookies=req.getCookies();
```
Comment (chamil321) [inline tests/jballerina-integration-test/src/test/resources/http/src/httpservices/43_http_cookies.bal:267]:
```suggestion
        http:Cookie[] reqstCookies = req.getCookies();
```

File: tests/jballerina-integration-test/src/test/resources/http/src/httpservices/43_http_cookies.bal [hunk 3]
```diff
@@ -258,4 +258,23 @@ service cookie on new http:Listener(9253) {
...
+        string meesage = "Valid cookies: ";
+        foreach http:Cookie cookie in  reqstCookies {
```
Comment (chamil321) [inline tests/jballerina-integration-test/src/test/resources/http/src/httpservices/43_http_cookies.bal:269]:
```suggestion
        foreach http:Cookie cookie in reqstCookies {
```

File: tests/jballerina-integration-test/src/test/resources/http/src/httpservices/43_http_cookies.bal [hunk 4]
```diff
@@ -258,4 +258,23 @@ service cookie on new http:Listener(9253) {
...
         }
     }
+
+    @http:ResourceConfig {
+        methods: ["GET"],
+        path: "/cookieBackend_13"
+   }
+    resource function validateCookie(http:Caller caller, http:Request req) {
+        http:Cookie[] reqstCookies=req.getCookies();
+        string meesage = "Valid cookies: ";
```
Comment (chamil321) [inline tests/jballerina-integration-test/src/test/resources/http/src/httpservices/43_http_cookies.bal:268]:
```suggestion
        string message = "Valid cookies: ";
```

File: tests/jballerina-integration-test/src/test/resources/http/src/httpservices/43_http_cookies.bal [hunk 5]
```diff
@@ -258,4 +258,23 @@ service cookie on new http:Listener(9253) {
...
+        path: "/cookieBackend_13"
+   }
+    resource function validateCookie(http:Caller caller, http:Request req) {
+        http:Cookie[] reqstCookies = req.getCookies();
+        string message = "Valid cookies: ";
+        foreach http:Cookie cookie in reqstCookies {
+            var value = cookie.value;
+            var name = cookie.name;
+            if (value is string && name is string) {
+                message = meesage.concat(name, "=", value , ",");
```
Comment (chamil321) [inline tests/jballerina-integration-test/src/test/resources/http/src/httpservices/43_http_cookies.bal:273]:
```suggestion
                message = mesage.concat(name, "=", value , ",");
```

File: stdlib/http/src/main/ballerina/src/http/cookie/cookie.bal (hunk 1 above)
Comment (kalaiyarasiganeshalingam) [reply stdlib/http/src/main/ballerina/src/http/cookie/cookie.bal:243]: Added in the #c6932c65a03c963879c258a151fc0d06b40d957e

File: stdlib/http/src/test/java/org/ballerinalang/stdlib/services/nativeimpl/request/RequestNativeFunctionSuccessTest.java [hunk 6]
```diff
@@ -748,4 +748,16 @@ public void testGetCookies() {
...
+    @Test
+    public void testGetCookiesWithEmptyValue() {
+        ObjectValue inRequest = createRequestObject();
+        ObjectValue entity = createEntityObject();
+        inRequest.set(REQUEST_ENTITY_FIELD, entity);
+        BValue[] returnVals = BRunUtil.invoke(compileResult, "testGetCookiesWithEmptyValue",
+                new Object[]{inRequest});
+        Assert.assertFalse(returnVals.length == 0 || returnVals[0] == null,
+                "No cookie objects in the Return Values");
+        Assert.assertEquals(returnVals.length, 1);
```
Comment (hemikak) [inline stdlib/http/src/test/java/org/ballerinalang/stdlib/services/nativeimpl/request/RequestNativeFunctionSuccessTest.java:761]: Isn't this assertion already implied by the previous assertion ?

File: tests/jballerina-integration-test/src/test/java/org/ballerinalang/test/service/http/sample/HTTPCookiesTestCase.java [hunk 7]
```diff
@@ -169,4 +169,15 @@ public void testSendPersistentCookiesWithoutPersistentCookieHandler() throws Bal
                 "cookieClient_12.bal"}, balFilePath);
         Assert.assertTrue(output.contains("SID003=895gd4dmnmsddd34"));
     }
+
+    @Test(description = "Test the cookie validation when using the getCookies()")
+    public void testCookieValidation() throws BallerinaTestException {
+        String balFilePath = (new File("src" + File.separator + "test" + File.separator + "resources" +
```
Comment (hemikak) [inline tests/jballerina-integration-test/src/test/java/org/ballerinalang/test/service/http/sample/HTTPCookiesTestCase.java:175]: We can use `Paths.get()` function here to create the path.

File: stdlib/http/src/test/java/org/ballerinalang/stdlib/services/nativeimpl/request/RequestNativeFunctionSuccessTest.java (hunk 6 above)
Comment (kalaiyarasiganeshalingam) [reply stdlib/http/src/test/java/org/ballerinalang/stdlib/services/nativeimpl/request/RequestNativeFunctionSuccessTest.java:761]: No, here, we are checking the actual length of the output but the previous assertion is checking that the output and output's first value are not empty.

**Pull request merged**